setup.sh

```diff
@@ -298,13 +298,23 @@ function synchronize_files()
   setup_config_file
   ASSERT_IF_NOT_EQ_ZERO "Config file failed" "$?"
 
-  if [ -f "$HOME/.bashrc" ]; then
-      # Add to bashrc
-      echo "# $app_name" >> "$HOME/.bashrc"
-      echo "source $libdir/$BASH_AUTOCOMPLETE.sh" >> "$HOME/.bashrc"
-      update_path
+  if [ -f "$HOME/.bashrc" ] || [ -h "$HOME/.bashrc" ]; then
+    # Add to bashrc
+    echo "# $app_name" >> "$HOME/.bashrc"
+    echo "source $libdir/$BASH_AUTOCOMPLETE.sh" >> "$HOME/.bashrc"
+    update_path
   else
-      warning "Unable to find a shell."
+    warning "Unable to find a bash shell."
+  fi
+
+  if [ -f "$HOME/.zshrc" ] || [ -h "$HOME/.zshrc" ]; then 
+    # Add to zshrc
+    echo "# Enable bash completion for zsh" >> "$HOME/.zshrc"
+    echo "autoload bashcompinit && bashcompinit" >> "$HOME/.zshrc"
+    echo "# $app_name" >> "$HOME/.zshrc"
+    echo "source $libdir/$BASH_AUTOCOMPLETE.sh" >> "$HOME/.zshrc"
+  else
+    warning "Unable to find a zsh shell."
```

Comment:
This code is very similar to the one above it; maybe you can avoid code duplication by making ".bashrc" and ".zshrc" values in a variable.

Reply:
I made another commit where i extracted these 2 lines into a function, let me know if this was what you had in mind. We could decrease the code duplication even further by extracting the whole tab completion process into a function, but I think this would unnecessarily increase complexity.